src/main/java/dev/the_fireplace/unforgivingvoid/usecase/VoidTransfer.java

```diff
@@ -41,15 +45,18 @@ public void initiateVoidTransfer(ServerPlayerEntity serverPlayerEntity, Minecraf
         DimensionConfig dimensionConfig = dimensionConfigManager.getSettings(currentWorld.getRegistryKey().getValue());
 
         ServerWorld targetWorld = getTargetWorld(server, dimensionConfig);
+
         if (targetWorld == null) {
             UnforgivingVoidConstants.getLogger().error("Target world not found: " + dimensionConfig.getTargetDimension());
             return;
         }
+
         spawnPositionLocator.setHorizontalOffsetRange(dimensionConfig.getHorizontalDistanceOffset());
         BlockPos spawnPos = getSpawnPos(serverPlayerEntity, currentWorld, dimensionConfig, targetWorld);
 
+        Entity teleportedEntity = teleporter.teleport(serverPlayerEntity, targetWorld, spawnPos.getX() + 0.5, spawnPos.getY(), spawnPos.getZ() + 0.5);
+
         applyStatusEffects(serverPlayerEntity, dimensionConfig);
```

Comment:
It might be safer to go ahead and applyStatusEffects to the teleported entity instead.
```suggestion
        applyStatusEffects(teleportedEntity, dimensionConfig);
```

Reply:
Will do